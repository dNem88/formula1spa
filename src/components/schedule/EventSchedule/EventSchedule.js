import React from 'react'
import styles from './eventSchedule.module.css'
import flags from '../../../assets/images/flags/flag-images'

function EventSchedule(props) {

    return (
        <section className={styles['main-container']}>
            <p className={styles['legend']}>ROUND 1</p>
            <div className={styles['date-container']}>
                <div className={styles['date-sub-container']}>
                    <span>23-25</span>
                    <span>FEB</span>
                </div>
                <div className={styles['flag-container']}>
                    <img src={flags['austria']} alt='national-flag' loading='lazy'/>
                </div>
            </div>
            <div className={styles['race-name-container']}>
                <div className={styles['race-sub-container']}>
                    <span>Austria</span>
                    <span>FORMULA 1 PRES-SEASON TRACK SESSION 2022</span>
                </div>
            </div>
            <div className={styles['image-container']}>
                <img src={'https://www.formula1.com/content/dam/fom-website/2018-redesign-assets/Track%20icons%204x3/Bahrain%20carbon.png.transform/2col/image.png'} alt='race-track-image' loading='lazy'/>
            </div>
        </section>
    )
}

export default EventSchedule