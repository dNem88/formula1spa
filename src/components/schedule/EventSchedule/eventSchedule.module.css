.main-container{
    box-sizing: border-box;
    position: relative;
    width: 23vw;
    height: 23vw;

    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
    align-items: center;
    justify-content: center;

    margin-top: 3vw;
    border-top: .2rem solid black;
    border-right: .2rem solid black;
    border-top-right-radius: 1rem;
}
.main-container:hover{
    border-top-color: var(--color-red);
    border-right-color: var(--color-red);
}
.legend{
    position: absolute;
    padding-top: 0;
    padding-bottom: 0;
    padding-left: 1vw;
    padding-right: 1vw;
    margin: 0;
    left: 0;
    top: -.7rem;
    height: clamp(.9rem, 1.2vw, 1.8rem);
    color: var(--color-red);
    font-size: clamp(.7rem, 1vw, 1.5rem);
    font-weight: bold;

    background-color: white;
}
.date-container{
    width: 100%;
    height: 25%;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
}
.date-sub-container{
    width: 60%;
    height: 100%;
    display: flex;
    flex-direction: column;
}
.date-sub-container > :nth-child(1) {
     background-color: white;
    color: black;
    width: max-content;
    font-size:clamp(1.2rem, 1.7vw, 2.5rem);
    font-weight: bold;
}
.date-sub-container > :nth-child(2) {
    background-color: black;
    color: white;
    width: max-content;
    font-size:clamp(1.2rem, 1.7vw, 2.5rem);
    font-weight: bold;
    padding-left: .5vw;
    padding-right: .5vw;
    border-radius: .5rem;
    display: flex;
    align-items: center;
    justify-content: center;
}
.flag-container{
    width: 15%;
    height: 38%;
    display: flex;
    align-items: center;
    justify-content: center;
    border: 1px solid grey;
    overflow: hidden;
    border-radius: .35rem;
    margin-right: 1vw;
}
.flag-container > :nth-child(1){
    width: 100%;
    height: 100%;
}


.race-name-container{
     width: 100%;
    height: 25%;
    border-top: .1rem solid black;
}
.race-sub-container{
    width: 100%;
    height: 100%;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
}
.race-sub-container > :nth-child(1){
    color: black;
    width: max-content;
    font-size:clamp(1.2rem, 1.7vw, 2.5rem);
    font-weight: bold;
}
.race-sub-container > :nth-child(1)::after{
    content:'>';
    color: var(--color-red);
    margin-left: 1vw;
}
.race-sub-container > :nth-child(2){
    background-color: white;
    color: black;
    font-size: clamp(.8rem, .75vw, 2rem);
}


.image-container{
    width: 100%;
    height: 50%;
    border-top: .1rem solid black;
}
.image-container > img{
    width: 100%;
    height: 100%;
}

@media only screen and (max-width: 600px)  {
    .main-container{
        width: 90vw;
        height: 90vw;   
    }
}
@media only screen and (mind-wdith: 601px) and(max-width: 600px)  {
    .main-container{
        width: 45vw;
        height: 45vw;   
    }
}
@media only screen and (max-width: 850px) and (orientation: landscape) {
    .main-container{
        width: 45vw;
        height: 45vw;   
}
}