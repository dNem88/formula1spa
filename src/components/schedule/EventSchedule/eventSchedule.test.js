import React from 'react'
import renderer from 'react-test-renderer'
import EventSchedule from './EventSchedule'
import {render, screen} from '@testing-library/react'

const mockProps = {
    round: '5',
    month: 'JAN',
    imageBase: 'italy',
    country: 'Italy',
    raceName: 'Test Grand Prix'
}


describe('Test ScheduleLayout',() => {
    it('Match Snapshot', () => {
        const snapshot = renderer.create(
            <EventSchedule {...mockProps}/>
        ).toJSON()
        expect(snapshot).toMatchSnapshot()
        expect(snapshot.props.className).toBe('main-container')
        expect(snapshot.children.length).toBe(4)
    })
    it('Elements text content', () => {
        render(<EventSchedule {...mockProps}/>)
        expect(screen.getByText('ROUND 5')).toBeTruthy()
        expect(screen.getByText('JAN')).toBeTruthy()
        expect(screen.getByText('Italy')).toBeTruthy()
        expect(screen.getByText('TEST GRAND PRIX', {exact: false})).toBeTruthy()
        expect(screen.getByAltText('race-track-image', {})).toBeTruthy()
    })
})