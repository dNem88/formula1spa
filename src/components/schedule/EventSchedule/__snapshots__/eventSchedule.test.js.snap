// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ScheduleLayout Match Snapshot 1`] = `
<section
  className="main-container"
>
  <p
    className="legend"
  >
    ROUND 
    5
  </p>
  <div
    className="date-container"
  >
    <div
      className="date-sub-container"
    >
      <span>
        
      </span>
      <span>
        JAN
      </span>
    </div>
    <div
      className="flag-container"
    >
      <img
        alt="national-flag"
        loading="lazy"
        src="italy.jpg"
      />
    </div>
  </div>
  <div
    className="race-name-container"
  >
    <div
      className="race-sub-container"
    >
      <span>
        Italy
      </span>
      <span>
        FORMULA 1 
        TEST GRAND PRIX
         2022
      </span>
    </div>
  </div>
  <div
    className="image-container"
  >
    <img
      alt="race-track-image"
      loading="lazy"
      src="italy.jpg"
    />
  </div>
</section>
`;
